uniro_frontend/src/components/customInput.tsx

```diff
@@ -1,35 +1,44 @@
-import { ChangeEvent, InputHTMLAttributes, useCallback, useState } from "react";
+import { ChangeEvent, InputHTMLAttributes, useCallback, useEffect, useRef, useState } from "react";
 import Search from "../../public/icons/search.svg?react";
 import ChevronLeft from "../../public/icons/chevron-left.svg?react";
 import Mic from "../../public/icons/mic.svg?react";
 import Close from "../../public/icons/close-circle.svg?react";
 
 interface CustomInputProps extends InputHTMLAttributes<HTMLInputElement> {
-	onLengthChange: (e: string) => void;
+	onChangeDebounce: (e: string) => void;
 	placeholder: string;
 	handleVoiceInput: () => void;
 }
 
-export default function Input({ onLengthChange, placeholder, handleVoiceInput, ...rest }: CustomInputProps) {
+export default function Input({ onChangeDebounce, placeholder, handleVoiceInput, ...rest }: CustomInputProps) {
 	const [isFocus, setIsFocus] = useState<boolean>(false);
-	const [tempValue, setTempValue] = useState<string>("");
-	const [currentValue, setCurrentValue] = useState<string>("");
+	const [value, setValue] = useState<string>("");
+	const timeOutRef = useRef<number>();
 
 	const onFoucs = () => setIsFocus(true);
 	const onBlur = () => setIsFocus(false);
 
 	const resetInput = () => {
-		setCurrentValue("");
-		setTempValue("");
+		setValue('');
+		handleDebounce('');
+	};
+
+	const handleDebounce = (input: string) => {
+		if (timeOutRef.current) {
+			clearTimeout(timeOutRef.current);
+		}
+
+		timeOutRef.current = setTimeout(() => {
+			onChangeDebounce(input);
+			timeOutRef.current = undefined;
+		}, 300);
 	};
 
```

Comment:
간단한 제안드립니다.
onchange 이후 timeout이 얼마 남아있지는 않지만, 안전한 로직을 위해 useEffect에 cleanup을 넣어주는건 어떨까요? 
```suggestion
useEffect(() => {
  return () => {
    if (timeOutRef.current) clearTimeout(timeOutRef.current);
  };
}, []);
```

Reply:
좋습니다! 추후 리팩토링에 반영하겠습니다